### ISSUE TITLE
JAX NUTS/ADVI sampling still fits corner-source ResistanceDistance, not the moving-window connectivity #105 standardized on

### ISSUE THREAD

**Issue by seansorek** [seansorek/DiffiScape#123]
## Description

The Bayesian sampling path was never migrated to the moving-window connectivity operator that GH #105 introduced for the optimizer, so NUTS and ADVI still fit against corner-source `ResistanceDistance` — the exact metric #105 removed everywhere else.

`inst/python/diffiscape_jax/sample.py`, `_build_numpyro_model`:

```python
        # Permeability conversion and circuit solve
        perm = prepare_permeability(surface_2d, parameterization)
        grid = GridGraph(grid=perm, fun=_mean_weight)
        dist_solver = ResistanceDistance()
        source = grid.coord_to_index(jnp.array([0]), jnp.array([0]))
        connectivity = dist_solver(grid, source)

        # PPP log-likelihood on valid cells with learnable alpha/gamma
        conn_valid = connectivity.ravel()[valid_mask]
        loglik = ppp_loglik(conn_valid, obs_jnp, cell_area, alpha, gamma)
```

Compare the point-estimate path, `core._connectivity_objective`, which after #105 reads:

```python
    permeability = prepare_permeability(surface_2d, parameterization)
    connectivity = cumulative_current_core(
        permeability, n_rows, n_cols, radius, block_size
    )
```

and `optimize.run_neural_optimization`'s `loss_fn`, which was migrated the same way.

`_build_numpyro_model` is the only likelihood used by both `run_nuts_sampling()` and `run_advi_sampling()`, i.e. by `ds_jax_sample_nuts()` and `ds_jax_sample_advi()`. Consistently, neither R wrapper even accepts `radius` / `block_size` — there is nowhere to pass the window geometry, because the sampler never uses one.

**Location:**
- [`inst/python/diffiscape_jax/sample.py:110-119`](https://github.com/seansorek/DiffiScape/blob/main/inst/python/diffiscape_jax/sample.py#L110-L119)
- Contrast: [`inst/python/diffiscape_jax/core.py:598-605`](https://github.com/seansorek/DiffiScape/blob/main/inst/python/diffiscape_jax/core.py#L598-L605)
- Callers: [`R/jax_bridge.R:498`](https://github.com/seansorek/DiffiScape/blob/main/R/jax_bridge.R#L498) (`ds_jax_sample_nuts`), [`R/jax_bridge.R:619`](https://github.com/seansorek/DiffiScape/blob/main/R/jax_bridge.R#L619) (`ds_jax_sample_advi`)

## Impact

This is #105's defect surviving in the one place the fix did not reach, and it is the same two failure modes:

1. **Different quantity, opposite sign convention.** `cumulative_current_core` accumulates moving-window current (high = well connected); a single-source `ResistanceDistance` map is a *distance* (high = poorly connected, and it is 0 at the source by construction). The posterior for `gamma` is therefore a coefficient on a different — roughly inversely related — covariate than the one `evaluate_full_model()`, `diffiscape()` step 6, `ds_posterior()` and `ds_jax_connectivity()` compute.
2. **Arbitrary source topology.** The whole landscape is scored relative to grid cell (0, 0), which for a masked study area is frequently an out-of-mask filler cell.

Concretely: the MAP/point estimate and the posterior are now fit to two different models. Posterior credible intervals on the resistance surface, and any downstream uncertainty propagation built on `samples_effective_loglinear`, describe a model that is never evaluated. Because the sampler still returns plausible-looking draws and summaries, nothing in the pipeline signals the mismatch.

## Suggested fix

Replace the `GridGraph`/`ResistanceDistance` block in `_build_numpyro_model` with the same `cumulative_current_core(perm, n_rows, n_cols, radius, block_size)` call the optimizer uses, thread `radius` / `block_size` through `run_nuts_sampling()` and `run_advi_sampling()`, and expose them on `ds_jax_sample_nuts()` / `ds_jax_sample_advi()` with the same defaults (13 / 5) as `optimize_resistance_gradient()`.

Note this will materially slow each likelihood evaluation (one windowed solve per leapfrog step rather than one global solve), which is worth confirming against `max_treedepth` before merging — but a fast wrong likelihood is not a useful trade here.

A regression test in the spirit of #105's would be: for a fixed surface, assert the sampler's internal connectivity equals `window.cumulative_current(...)["current"]` rather than merely being finite.

---
*Auto-detected by codebase-review agent*

**Comment by seansorek** [comment]
**Triage: Confirmed bug — high severity (posterior incoherent with forward path)**

**Bug confirmed.** The sampling path (`ds_jax_sample_nuts` / `ds_jax_sample_advi`) was never updated when GH #105 replaced the point-estimate optimizer's single-source `ResistanceDistance` with the moving-window `cumulative_current_core` operator. As a result, NUTS and ADVI fit the posterior against a corner-source resistance map (rooted at grid cell (0,0)), while `evaluate_full_model` / `ds_jax_connectivity` evaluate the moving-window quantity. The fitted `gamma` values from sampling have no consistent interpretation relative to the optimizer or the forward evaluation path — the two quantities are inversely related for landscape-scale resistance problems.

**Severity: high.** Any workflow that (1) fits parameters via sampling and (2) evaluates or predicts using `ds_jax_connectivity` / `evaluate_full_model` is silently using an incoherent gamma. No data corruption or crash, but scientific results would be wrong.

**Fix:** Open PR #133 (`fix/issue-123`) replaces `GridGraph + ResistanceDistance` with `cumulative_current_core` inside `_build_numpyro_model`. It adds `radius`/`block_size` params defaulting to 13/5 (matching `optimize_resistance_gradient`'s defaults) and appends them at the end of the R and Python signatures to preserve positional-call compatibility. CI is passing on the current head. The PR will be merged by the auto-merge workflow once marked Reviewed.

**Comment by seansorek** [comment]
Fixed by #133.
